### PR TITLE
[RHELC-871] Redefine test tiers based on criticality

### PR DESCRIPTION
**Pull request by danmyway**
* rename directories tier0 -> non-destructive, tier1 -> destructive
* redefine tier0 and tier1 plans based on test criticality
* tier0 to run for each PR
* tier1 to run for each merge commit to main branch

Jira Issues: [RHELC-871](https://issues.redhat.com/browse/RHELC-871)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: plans/tier0.fmf [hunk 1]
```diff
@@ -1,9 +1,200 @@
...
+                  how: shell
+                  script: pytest -svv tests/integration/destructive/method/rhsm.py
+                - name: reboot after conversion
+                  how: ansible
+                  playbook: tests/ansible_collections/roles/reboot/main.yml
+
+        /rhsm_non_eus:
+            adjust+:
+                - enabled: false
+                  # TODO EUS Bump disabled version
```
[the remark this answers is not in the document]
Comment (danmyway) [reply plans/tier0.fmf:42]: When we get back to supported EUS paths.
At this moment that would be only Rocky/Alma (8.6) relevant, however the full coverage/support of those targets is currently out of scope.

[the remark this answers is not in the document]
Comment (danmyway) [reply plans/tier0.fmf:42]: So for now this plan would get disabled

File: tests/integration/tier1/non-destructive/main.fmf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+summary+: |
+    Set of non-destructive tests
+
+description+: |
+    All tests in tier0 are non-destructive, meaning there is not a full conversion being done.
```
[the remark this answers is not in the document]
Comment (danmyway) [reply tests/integration/tier1/non-destructive/main.fmf:5]: I think I will remove the mention of tiers here completely, it's not really relevant to the destructive/non-destructive context anyway.

File: plans/tier1.fmf [hunk 3]
```diff
@@ -1,73 +1,16 @@
...
-          how: shell
-          script: pytest -svv tests/integration/tier1/method/custom_repos.py
-        - name: reboot after conversion
-          how: ansible
-          playbook: tests/ansible_collections/roles/reboot/main.yml
+summary+: |
+    Tier1 tests
+description+: |
+    Set of tests/plans of less frequency demands
+    for each merge commit to the main branch.
```
[the remark this answers is not in the document]
Comment (danmyway) [reply plans/tier1.fmf:6]: The parent label is in `tests/integration/tier0/main.fmf` and `tests/integration/tier1/main.fmf`

**Review by kokesak — APPROVED** [review]
All tests passed except the offline centos7 satellite, which failed on not synced repository containing an older kernel. 
The changes in the PR look good.

**Pull request merged**